lib/builtin.t

```diff
@@ -0,0 +1,39 @@
+#!./perl
+
+BEGIN {
+    chdir 't' if -d 't';
+    require './test.pl';
+    set_up_inc('../lib');
+}
+
+use strict;
+use warnings;
+
+# booleans
+{
+    use builtin qw( true false isbool );
+
+    ok(true, 'true is true');
+    ok(!false, 'false is false');
+
+    ok(isbool(true), 'true is bool');
+    ok(isbool(false), 'false is bool');
+    ok(!isbool(undef), 'undef is not bool');
+    ok(!isbool(1), '1 is not bool');
+    ok(!isbool(""), 'empty is not bool');
+
+    my $truevar  = (5 == 5);
+    my $falsevar = (5 == 6);
+
+    ok(isbool($truevar), '$truevar is bool');
+    ok(isbool($falsevar), '$falsevar is bool');
+}
```

Comment:
Since the result of `isbool` is also a boolean, shall we include these tests as proofs ?

```perl

ok( isbool(isbool(true)), 'isbool() returns a bool' );
ok( isbool(isbool(false)), 'isbool() returns a bool' );
ok( isbool(isbool(undef)), 'isbool() returns a bool' );
ok( isbool(isbool(1)), 'isbool() returns a bool' );
ok( isbool(isbool("")), 'isbool() returns a bool' );
ok( isbool(isbool($truevar)), 'isbool() returns a bool' );
ok( isbool(isbool($false)), 'isbool() returns a bool' );
```

Reply:
Ahyes, a good idea. I don't think it's necessary to test all the possible combinations there, I'll just do one true and one false case.

Added in a4837515be